app/src/pages/inside/common/modals/retentionPolicy/hooks.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useDispatch } from 'react-redux';
+import { defineMessages, useIntl } from 'react-intl';
+import { fetch } from 'common/utils';
+import { URLS } from 'common/urls';
+import { RETENTION_POLICY } from 'common/constants/retentionPolicy';
+import { NOTIFICATION_TYPES, showNotification } from 'controllers/notification';
+
+const messages = defineMessages({
+  successNotification: {
+    id: 'RetentionPolicy.successNotification',
+    defaultMessage: 'Launch has been updated successfully',
+  },
+});
+
+export const useFetchRetentionPolicy = (
+  isImportant,
+  activeProject,
+  launch,
+  updateLaunchLocally,
+) => {
+  const dispatch = useDispatch();
+  const { formatMessage } = useIntl();
+
+  const policy = isImportant ? RETENTION_POLICY.IMPORTANT : RETENTION_POLICY.REGULAR;
+
+  const fetchRetentionPolicy = () => {
+    fetch(URLS.singleLaunchUpdate(activeProject, launch.id), {
+      method: 'PUT',
+      data: [
+        {
+          key: RETENTION_POLICY.RETENTION_POLICY_KEY,
+          value: policy,
+          system: true,
+        },
+      ],
```

Comment:
I see the 'bad request' message after sending the request.
![image](https://github.com/reportportal/service-ui/assets/31861805/8ac1cd87-3d06-4c63-8385-f3c0adb4f189)
Did you check that it work?
I suppose the request should look like this:
```javascript
const data = {
   attributes: [
        {
          key: RETENTION_POLICY.RETENTION_POLICY_KEY,
          value: policy,
          system: true,
        },
      ],
};
```

According to API Docs

![image](https://github.com/reportportal/service-ui/assets/31861805/774eede9-3d18-40f7-8b29-17879ec4359c)